Gruntfile.js

```diff
@@ -1219,22 +1219,32 @@ module.exports = function(grunt) {
 	] );
 
 	grunt.registerTask( 'sync-gutenberg-packages', function() {
-		if ( grunt.option( 'update-browserlist' ) ) {
-			/*
-			 * Updating the browserlist database is opt-in and up to the release lead.
-			 *
-			 * Browserlist database should be updated:
-			 * - In each release cycle up until RC1
-			 * - If Webpack throws a warning about an outdated database
-			 *
-			 * It should not be updated:
-			 * - After the RC1
-			 * - When backporting fixes to older WordPress releases.
-			 *
-			 * For more context, see:
-			 * https://github.com/WordPress/wordpress-develop/pull/2621#discussion_r859840515
-			 * https://core.trac.wordpress.org/ticket/55559
-			 */
+		if ( ! grunt.option( 'dev' ) ) {
+			console.log( 'You must manually pass the --dev flag to the sync-gutenberg-packages task as follows:' );
+			console.log( 'npx grunt sync-gutenberg-packages --dev' );
+			console.log( 'Otherwise the webpack build tasks are executed in the production mode and do not ');
+			console.log( 'regenerate the src/wp-includes/assets/script-loader-packages.php.' );
+			process.exit( 0 );
+		}
+
+		/*
+		 * Browserlist database should be updated when the packages are
+		 * being updated to their latest version in the trunk branch.
+		 *
+		 * It should not happen during the patch releases. Otherwise,
+		 * we'd be changing the supported browsers of a WP release on patch release.
+		 *
+		 * For more context, see:
+		 *
+		 * https://github.com/WordPress/gutenberg/issues/33344
+		 * https://core.trac.wordpress.org/ticket/55559
+		 */
+		const distTag = grunt.option('dist-tag') || 'latest';
+		const currentBranch = spawn( 'git', [ 'branch', '--show-current' ], {
```

Comment:
It won't work for svn users 😅 

However, I think it's very unlikely you would run `latest` outside of the `trunk` branch.

Reply:
Let's not block this PR on that and move this discussion to a separate one